src/main/java/com/exactpro/th2/common/event/Event.java

```diff
@@ -205,11 +207,27 @@ public Event messageID(MessageID attachedMessageID) {
         return this;
     }
 
-    public List<com.exactpro.th2.common.grpc.Event> toProtoEvents(String parentID) throws JsonProcessingException {
+    /**
+     * @deprecated prefer to use full object instead of part of them, use the {@link #toProtoEvents(EventID)} method
+     */
+    public List<com.exactpro.th2.common.grpc.Event> toProtoEvents(@Nullable String parentID) throws JsonProcessingException {
+        return toProtoEvents(toEventID(parentID));
+    }
+
+
+    public List<com.exactpro.th2.common.grpc.Event> toProtoEvents(@Nullable EventID parentID) throws JsonProcessingException {
         return collectSubEvents(new ArrayList<>(), parentID);
     }
 
+    /**
+     * @deprecated prefer to use full object instead of part of them, use the {@link #toProtoEvent(EventID)} method
+     */
+    @Deprecated
     public com.exactpro.th2.common.grpc.Event toProtoEvent(@Nullable String parentID) throws JsonProcessingException {
+        return toProtoEvent(toEventID(parentID));
+    }
+
+    public com.exactpro.th2.common.grpc.Event toProtoEvent(@Nullable EventID parentID) throws JsonProcessingException {
```

Comment:
This new method will break the compatibility. If the original code calls `toProtoEvent(String)` with `null` parameter after adding that method the code won't compile. The compile can't choose between those two methods.

If it is okay and we are ready to change some places in code if we update the common-j version then ok, let's submit those changes